src/site/markdown/server/manage-webadmin.md

```diff
@@ -2561,23 +2561,126 @@ Deleted messages of a specific user can be restored by calling the following end
+      "fieldName": "originMailboxes",
+      "operator": "contains",
+      "value":  "02874f7c-d10e-102f-acda-0015176f7922"
+    }
+  ]
+};
+```
+
+The requested Json body is made from list of criterion objects which have following structure:
+```
+{
+  "fieldName": "supportedFieldName",
+  "operator": "supportedOperator",
+  "testedValue": "plain string represents for the matching value of corresponding field"
+}
 ```
+Deleted Messages which are matched with **all** criterions in the query body will be restored. Here are list of supported fieldName for the restoring:
+ - subject: represents for deleted message `subject` field matching. Supports below string operators:
+   - contains
+   - containsIgnoreCase
+   - equals
+   - equalsIgnoreCase
+ - deliveryDate: represents for deleted message `deliveryDate` field matching. Tested value should follow the right date time with zone offset format (ISO-8601) like
+   `2008-09-15T15:53:00+05:00` or `2008-09-15T15:53:00Z` 
+   Supports below date time operators:
+   - beforeOrEquals: is the deleted message's `deliveryDate` before or equals the time of tested value.
+   - afterOrEquals: is the deleted message's `deliveryDate` after or equals the time of tested value
+ - deletionDate: represents for deleted message `deletionDate` field matching. Tested value & Supports operators: similar to `deliveryDate`
+ - sender: represents for deleted message `sender` field matching. Tested value should be a valid mail address. Supports mail address operator:
+   - equals: does the tested sender equal to the sender of the tested deleted message ?   
+ - recipients: represents for deleted message `recipients` field matching. Tested value should be a valid mail address. Supports list mail address operator:
+   - contains: does the tested deleted message's recipients contain tested recipient ?
+ - hasAttachment: represents for deleted message `hasAttachment` field matching. Tested value could be `false` or `true`. Supports boolean operator:
+   - equals: does the tested deleted message's hasAttachment property equal to the tested hasAttachment value?
+ - originMailboxes: represents for deleted message `originMailboxes` field matching. Tested value is a string serialized of mailbox id. Supports list mailbox id operators:
+   - contains: does the tested deleted message's originMailbox ids contain tested mailbox id ?
+   
+Messages in the Deleted Messages Vault of an specified user that are matched with Query Json Object in the body will be appended to his 'Restored-Messages' mailbox, which will be created if needed.
 
-**All** messages in the Deleted Messages Vault of an specified user will be appended to his 'Restored-Messages' mailbox, which will be created if needed.
+**Note**:
+ - fieldName & operator for passing to the routes are case sensitive
+ - Currently, we only support query combinator `and` value, otherwise, requests will be rejected 
+ - Query could have a nested structure where criteria contains queries:
```

Comment:
` Query could have a nested structure where criteria contains queries:`

Not true yet, we should not document something you spend loads of energy saying it don't work yet below.

Please say that only `flat and` is supported now (as we don't plan anything else anysoon)

Reply:
At the lines below, I had an explaination 
```
But at the current version, this structure is not allowed to be handled. You should go with the flatten structure
```